src/app/file-manager/transferables-grid/transferables-grid.component.ts

```diff
@@ -1,54 +1,147 @@
-import { Component, OnInit, ViewChild, ChangeDetectionStrategy } from '@angular/core';
+import { Component, OnInit, AfterViewInit, Input, ViewChild, ChangeDetectionStrategy } from '@angular/core';
 
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs/Observable';
 import * as Downloadables from '../actions/downloadables.actions';
 import { Item } from '../models/item';
 import { DownloadableState, DownloadablesReducer } from '../reducers/downloadables.reducer';
-import { MatPaginator, PageEvent } from '@angular/material';
-import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator, MatSort, PageEvent } from '@angular/material';
+import { animate, state, style, transition, trigger } from '@angular/animations';
+
+import { DataSource, CollectionViewer } from '@angular/cdk/collections';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/startWith';
+import 'rxjs/add/observable/forkJoin';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/merge';
 
 interface AppState {
   downloadables: DownloadableState;
 }
 
+export class FilesDatabase {
+  itemsObs: Observable<DownloadableState>;
+  totalCount = 0;
+
+  dataChange: BehaviorSubject<Item[]> = new BehaviorSubject<Item[]>([]);
+  selectionChange: BehaviorSubject<number> = new BehaviorSubject<number>(0);
+
+  get data(): Item[] {
+    return this.dataChange.value;
+  }
+
+  get selectedCount(): number {
+    return this.selectionChange.value;
+  }
+
+  constructor(private store: Store<AppState>) {
+
+    this.itemsObs = store.select('downloadables');
+    this.itemsObs.subscribe(
+      data => {
+        this.dataChange.next(data.items);
+        this.selectionChange.next(data.selectedCount);
+        this.totalCount = data.count;
+      }
+    );
+  }
+}
+
+export class FilesDataSource extends DataSource<Item> {
+  itemsObs: Observable<DownloadableState>;
+  dataChange: BehaviorSubject<Item[]> = new BehaviorSubject<Item[]>([]);
+
+  constructor(private filesDB: FilesDatabase,
+    private store: Store<AppState>,
+    private _sort: MatSort,
+    private _paginator: MatPaginator) {
+
+    super();
+  }
+
+  connect(): Observable<Item[]> {
+
+    const displayDataChanges = [
+      this.filesDB.dataChange,
+      this._paginator.page,
+      this.filesDB.selectionChange,
+      // this._sort.sortChange
+    ];
+
+    return Observable.merge(...displayDataChanges).map(() => {
+
+      // const data = this.getSortedData(this.filesDB.data.slice());
+      const data = this.filesDB.data.slice();
+
+      // Grab the page's slice of data.
+      const startIndex = this._paginator.pageIndex * this._paginator.pageSize;
+      return data.splice(startIndex, this._paginator.pageSize);
+    });
+  }
+
+  disconnect() {
+
+  }
+
+  getSortedData(data: Item[]): Item[] {
```

Comment:
It looks like this is prototype code that is currently unused. In general it's better to not commit code until it's used, to avoid having to clean up code later on.